fix(micro-default-editor): Only call DNF once for update script

GildedRoach opened the pull request.

GildedRoach committed:
fix(micro-default-editor): Only call DNF once for update script

Signed-off-by: Gilver <rockgrub@disroot.org>

madonuko commented:
```ts
let v = sh("dnf rq --qf='%{version}|%{release}|%{epoch}' micro --repo=fedora,updates", #{"stdout": "piped"}).ctx.stdout;
let vre = v.split("|");
rpm.version(vre[0]);
print(vre[1].split(".")[0].parse_int());
rpm.release(vre[1].split(".")[0].parse_int());
rpm.f = sub(`^Epoch: (.+?)$`, "Epoch: " + vre[2], rpm.f);
```
is this better?

GildedRoach commented:
> ```ts
> let v = sh("dnf rq --qf='%{version}|%{release}|%{epoch}' micro --repo=fedora,updates", #{"stdout": "piped"}).ctx.stdout;
> let vre = v.split("|");
> rpm.version(vre[0]);
> print(vre[1].split(".")[0].parse_int());
> rpm.release(vre[1].split(".")[0].parse_int());
> rpm.f = sub(`^Epoch: (.+?)$`, "Epoch: " + vre[2], rpm.f);
> ```

This works

GildedRoach committed:
Funny VRE

Signed-off-by: Gilver <rockgrub@disroot.org>

madonuko reviewed (approved):
approve my own script yay

raboneko commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|f41|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/5844">](https://github.com/terrapkg/packages/pull/5844)|
|✅|el10|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/5845">](https://github.com/terrapkg/packages/pull/5845)|
|✅|f42|[<img src="https://img.shields.io/github/pulls/detail/state/terrapkg/packages/5846">](https://github.com/terrapkg/packages/pull/5846)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport) and see the [Github Action logs](https://github.com/terrapkg/packages/actions/runs/16263541214) for details

<!--- Backport version: 9.5.1 -->
